src/test/java/de/mediathekview/mserver/crawler/phoenix/tasks/PhoenixFilmDetailTaskTest.java

```diff
@@ -0,0 +1,137 @@
+import de.mediathekview.mserver.base.config.MServerConfigManager;
+import de.mediathekview.mserver.crawler.basic.CrawlerUrlDTO;
+import de.mediathekview.mserver.crawler.phoenix.PhoenixCrawler;
+import de.mediathekview.mserver.progress.listeners.SenderProgressListener;
+import de.mediathekview.mserver.testhelper.AssertFilm;
+import de.mediathekview.mserver.testhelper.WireMockTestBase;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ForkJoinPool;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class PhoenixFilmDetailTaskTest extends WireMockTestBase {
+
+  @Parameters
+  public static Collection<Object[]> data() {
+    return Arrays.asList(new Object[][]{
+        {
+            "/response/id/271252",
+            "/phoenix/phoenix_film_detail1.json",
+            "/php/mediaplayer/data/beitrags_details.php?ak=web&ptmd=true&id=293872",
+            "/phoenix/phoenix_film_detail1.xml",
+            "/tmd/2/ngplayer_2_3/vod/ptmd/phoenix/180624_phx_presseclub",
+            "/phoenix/phoenix_video_detail1.json",
+            "Presseclub",
+            "Mehr Grenzschutz und eine neue Asylpolitik – letzte Rettung für Europa und Merkel?",
+            LocalDateTime.of(2018, 6, 24, 12, 0, 0),
+            Duration.ofMinutes(57).plusSeconds(12),
+            "Moderation: Sonia Seymour Mikich",
+            "https://www.phoenix.de/sendungen/gespraeche/presseclub/mehr-grenzschutz-und-eine-neue-asylpolitik--letzte-rettung-fuer-europa-und-merkel-a-271252.html",
+            "https://rodlzdf-a.akamaihd.net/none/phoenix/18/06/180624_phx_presseclub/1/180624_phx_presseclub_776k_p11v13.mp4",
+            "https://rodlzdf-a.akamaihd.net/none/phoenix/18/06/180624_phx_presseclub/1/180624_phx_presseclub_2328k_p35v13.mp4",
+            "",
+            "",
+            GeoLocations.GEO_NONE
+        }
+    });
+  }
+
+  private final String filmUrl;
+  private final String filmJsonFile;
+  private final String filmDetailXmlUrl;
+  private final String filmDetailXmlFile;
+  private final String videoUrl;
+  private final String videoJsonFile;
+  private final String expectedTopic;
+  private final String expectedTitle;
+  private final LocalDateTime expectedTime;
+  private final Duration expectedDuration;
+  private final String expectedDescription;
+  private final String expectedWebsite;
+  private final String expectedUrlSmall;
+  private final String expectedUrlNormal;
+  private final String expectedUrlHd;
+  private final String expectedSubtitle;
+  private final GeoLocations expectedGeo;
+
+  public PhoenixFilmDetailTaskTest(final String aFilmUrl, final String aFilmJsonFile,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 17 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)